Fix watch storefront multi saleschannel with multi theme

raffaelecarelle opened the pull request:
Implemented prompts to select a theme if multiple exist and a domain URL if multiple domains are associated with the theme. These changes improve user experience in handling multi-theme and multi-sales channel setups. Updated `start-hot-reload.js` to use the selected domain URL.

### 1. Why is this change necessary?
These changes improve user experience in handling multi-theme and multi-sales channel webpack watching development 

### 2. What does this change do, exactly?
Implemented prompts to select a theme if multiple exist and a domain URL if multiple domains are associated with the theme. Updated `start-hot-reload.js` to use the selected domain URL.

### 3. Describe each step to reproduce the issue or behaviour.
Add more then one sales channel and more than one theme. theme:dump command dumps only first theme that it can find in the system.

### 4. Please link to the relevant issues (if any).
Fixes #4133

raffaelecarelle committed:
Update ThemeDumpCommand to prompt for themes and domains

Implemented prompts to select a theme if multiple exist and a domain URL if multiple domains are associated with the theme. These changes improve user experience in handling multi-theme and multi-sales channel setups. Updated `start-hot-reload.js` to use the selected domain URL.

-----------------------------------------------------------------

Refactor ThemeDumpCommand for improved argument handling

Enhanced the ThemeDumpCommand to include support for optional 'domain-url' and improved handling of 'theme-id'. Updated test cases to cover the new argument scenarios and added user prompts when arguments are missing. This refactor ensures more robust and user-friendly command execution.

runelaenen commented:
@raffaelecarelle Should it also not be possible to add the theme and domain as arguments to the command? Currently the command becomes unusable in CI pipelines because it can only be finished interactively when there are multiple domains or themes.

raffaelecarelle committed:
Changed theme filtering from 'technicalName' to 'name' in ThemeDumpCommand. This because theme created from administration panel, don't have technical name.

raffaelecarelle commented:
> @raffaelecarelle Should it also not be possible to add the theme and domain as arguments to the command? Currently the command becomes unusable in CI pipelines because it can only be finished interactively when there are multiple domains or themes.

Good point.

I've added the domain-url argument (theme-id was already in place) and test case for Command and some adjustements. 

Thanks!

raffaelecarelle commented:
Hello @shyim, can you review this PR? I think it can be useful. Thank you.

BrocksiNet commented:
Should it also be possible to change the SalesChannel? Or did I overlook something?

raffaelecarelle committed:
Remove unused PHPUnit import

The `CoversClass` import was removed from `ThemeDumpCommandTest.php` as it was no longer being used. This cleanup improves code readability and reduces unnecessary dependencies.

BrocksiNet commented:
Hi @raffaelecarelle, it took quite some time because we also had changed `start-hot-reload.js` (https://github.com/shopware/shopware/commit/46c8d8b44969dba042165da0b2af373ca895d28e).  
Anyway, we tested and integrated your idea into the code, and also added some simple unit tests. 

Thx for your contribution. :clap:
